src/Libraries/RevitNodes/GeometryConversion/DynamoToRevitBRep.cs

```diff
@@ -0,0 +1,150 @@
+       /// <param name="sol"></param>
+       /// <param name="performHostUnitConversion"></param>
+       /// <returns></returns>
+       public static GeometryObject ToRevitType(Solid sol,
+          bool performHostUnitConversion = true)
+       {
+            var geometry = ToRevitType(sol, performHostUnitConversion, BRepType.Solid);
+
+           if (geometry == null)
+           {
+               // An unexpected failure occurred when attempting to convert the solid into a Revit BRep.
+               throw new Exception(Properties.Resources.DynamoSolidToRevitBRepFailure);
+           }
+
+            return geometry;
+       }
+
+       /// <summary>
+       /// this method attmempts to construct a BRep from a surface.
+       /// </summary>
+       /// <param name="surf"></param>
+       /// <param name="performHostUnitConversion"></param>
+       /// <returns></returns>
+       public static GeometryObject ToRevitType(Surface surf,
+          bool performHostUnitConversion = true)
+       {
+            var geometry = ToRevitType(surf, performHostUnitConversion, BRepType.OpenShell);
+
+            if (geometry == null)
+            {
+                // An unexpected failure occurred when attempting to convert the surface into a Revit BRep
+                throw new Exception(Properties.Resources.DynamoSurfaceToRevitBRepFailure);
+            }
+
+           return geometry;
+       }
+
+       private static GeometryObject ToRevitType(Autodesk.DesignScript.Geometry.Topology topology,
+       bool performHostUnitConversion,
+       BRepType type)
+       {
+           List<Face> faces = null;
+
+           if (performHostUnitConversion)
+           {
+               using(var newTopology = topology.InHostUnits())
+               {
+                   faces = newTopology.Faces.ToList();
+               }
+           }
+           else
+           {
+               faces = topology.Faces.ToList();
+           }
+
+           var brb = new BRepBuilder(type);
+           var edge2EdgeId = new Dictionary<Edge, BRepBuilderGeometryId>();
+
+           //foreach face in solid/surface
+           foreach (Face protoFace in faces)
+           {
+               using(var geom = protoFace.SurfaceGeometry())
+               {
+                   using(var ngeom = geom.ToNurbsSurface())
+                   {
+                       bool flipped = false;
+                       // Check if the nurbs surface has flipped compared to the original surface
+                       if (geom.NormalAtParameter(.5, .5).Dot(ngeom.NormalAtParameter(.5, .5)) < 0)
+                       {
+                           flipped = true;
+                       }
+
+                       // Create Revit nurbs surface
+                       var bbface = BRepBuilderSurfaceGeometry.CreateNURBSSurface(ngeom.DegreeU, ngeom.DegreeV,
+                           ngeom.UKnots(), ngeom.VKnots(), ngeom.ControlPoints().SelectMany(x => x.Select(y => y.ToXyz())).ToList(),
```

Comment:
@sm6srw I'm still a bit worried about the XYZs heres, have you thought about if we need to convert them to the host units?